docs/references/expo/expo-mfa.mdx

```diff
@@ -0,0 +1,391 @@
+    1. In the **Authentication strategies** section of this page, ensure **Password** is enabled.
+1. You must also configure your Clerk instance to enable multi-factor authentication:
+    1. Navigate to your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/multi-factor) and in the navigation sidebar, select **User & Authentication > Multi-factor**.
+    1. For this example, enable the **Authenticator application** and **Backup codes** strategies.
+1. Install the `expo-checkbox` package for the UI:
+    <CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+      ```bash {{ filename: "terminal" }}
+      npm install expo-checkbox
+      ```
+      ```bash {{ filename: "terminal" }}
+      yarn add expo-checkbox
+      ```
+      ```bash {{ filename: "terminal" }}
+      pnpm add expo-checkbox
+      ```
+    </CodeBlockTabs>
+
+## Create an MFA component
+
+The following example demonstrates a full MFA sign-in flow that enables users to generate QR codes which can be used with an authenticator app.
+
+> You can render this component in a custom sign-in or sign-up flow, which you can find examples of in [the Expo quickstart](/docs/quickstarts/expo).
+
+<Steps>
+
+### Create the sign-in form
+
+In your components directory, create the MFA sign-in form.
+
+```tsx {{ filename: "components/SignInMFAForm.tsx" }}
+import React from "react";
+import { useSignIn } from "@clerk/clerk-expo";
+import { useRouter } from "expo-router";
+import { Text, TextInput, Button, View } from "react-native";
+import Checkbox from "expo-checkbox";
+
+export default function SignInMFAForm() {
+  const { signIn, setActive, isLoaded } = useSignIn();
+
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [code, setCode] = React.useState("");
+  const [useBackupCode, setUseBackupCode] = React.useState(false);
+  const [displayTOTP, setDisplayTOTP] = React.useState(false);
+  const router = useRouter();
+
+  const handleFirstStage = () => setDisplayTOTP(true);
+
+  const onPressTOTP = React.useCallback(async () => {
+    if (!isLoaded) return;
+
+    // Start the sign-in process
+    try {
+      const attemptFirstFactor = await signIn.create({
+        identifier: email,
+        password,
+      });
+
+      // Attempt to sign in the user with email / password incase they haven't yet activated TOTP & Backup codes
+      if (attemptFirstFactor.status === "complete") {
+
+        // If the attempt is successful, re-route them past the MFA step to the dashboard
+        await setActive({ session: attemptFirstFactor.createdSessionId })
+
+        router.replace("/dashboard")
+      }
+
+      // Attempt the TOTP or backup code verification
+      const attemptSecondFactor = await signIn.attemptSecondFactor({
+        strategy: useBackupCode ? "backup_code" : "totp",
+        code: code,
+      });
```

Comment:
yeah, I had added the `else` check but I think Dom removed it because it kept throwing an error? 

@domitriusclark essentially, there should be this after the if (attemptFirstFactor....)
```
else {
        // If the status is not complete, check why. User may need to
        // complete further steps.
        console.error(JSON.stringify(attemptSecondFactor,null,2));
}
```

but I think the entire `if (attemptFirstFactor...)` logic should be moved to `handleFirstStage()`

that might be why you kept getting an error @domitriusclark

Reply:
I think the error was happening because if it was anything else _but_ `complete` it would catch _then_ hit the TOTP flow. 

I'll mess around with some better error handling for these specific cases 👍